### PR TITLE
(SERVER-319) Apache-style request logging

### PR DESCRIPTION
**Pull request by nwolfe**
This commit adds a new Apache-style puppetserver-access.log file to
capture HTTP traffic.

### CODE REVIEW

**Comment by fpringvaldsen** [comment]
I've just tested on CentOS 7, and it looks like the configuration is pointing to the access logging file, but that `request-logging.xml` is not appearing in the `/etc/puppetserver/` directory.

**Comment by nwolfe** [comment]
@fpringvaldsen It's in /var/log/puppetserver with the other log files

**Comment by fpringvaldsen** [comment]
@nwolfe I'm talking about the `request-logging.xml` configuration file itself. The `webserver.conf` file is setting the access log configuration to `/etc/puppetserver/request-logging.xml`, but that configuration file does not exist, so access logging is not happening.

**Comment by nwolfe** [comment]
@fpringvaldsen I've double-checked this work on a couple VMs and things are working as expected. I think you might need to make sure the .jar is deleted from your ~/.m2 and do a fresh `lein install` before building the package locally.

**Comment by cprice404** [comment]
+1, one question though: would it be worth plumbing this stuff into the example files for devs as well?  Seems easy-ish and probably valuable?

**Comment by nwolfe** [comment]
@cprice404 Yeah good idea, doing that now...

**Comment by fpringvaldsen** [comment]
@nwolfe Confirmed that it works after doing a `lein install`, sorry about that! Thanks for your help. I have now functionally reviewed and confirmed that this is working so I'm :+1: barring putting this into the example files for devs.

File: dev/request-logging-dev.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<configuration scan="true" debug="false">
+  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <encoder>
+        <pattern>%h %l %u %user %date "%r" %s %b %h %a %localPort %D</pattern>
+    </encoder>
+  </appender>
+  <appender-ref ref="STDOUT" />
+</configuration>
```
Comment (nwolfe) [inline dev/request-logging-dev.xml]: @cprice404 Let me know if this is what you were thinking

**Comment by nwolfe** [comment]
Updated with dev/source support

File: dev/request-logging-dev.xml (hunk 1 above)
Comment (cprice404) [inline dev/request-logging-dev.xml]: Pretty much, except I hadn't been thinking we would dump that stuff to STDOUT.  I think I was thinking maybe we'd use a file appender, but hadn't thought much about where the file would go.  Maybe just in like './target/logs/puppetserver-access.log' or something?

Comment (nwolfe) [inline dev/request-logging-dev.xml]: Yeah, having the output mixed in with the other STDOUT looked pretty bad.  Will move it to ./target

File: dev/request-logging-dev.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<configuration scan="true" debug="false">
+  <appender name="FILE" class="ch.qos.logback.core.FileAppender">
+    <file>./target/logs/puppetserver-access.log</file>
+    <encoder>
+        <pattern>%h %l %u %user %date "%r" %s %b %h %a %localPort %D</pattern>
+    </encoder>
+  </appender>
+  <appender-ref ref="FILE" />
+</configuration>
```
Comment (nwolfe) [inline dev/request-logging-dev.xml]: Updated to write to a file instead of STDOUT.

**Pull request merged**